tests/FakeItEasy.Tests/ArgumentConstraintTestBase.cs

```diff
@@ -1,43 +1,48 @@
 namespace FakeItEasy.Tests
 {
+    using System;
+    using System.Collections.Generic;
     using System.Text;
-    using FakeItEasy.Core;
+    using FakeItEasy.Tests.ArgumentConstraintManagerExtensions;
     using FakeItEasy.Tests.TestHelpers;
     using FluentAssertions;
     using Xunit;
 
-    public abstract class ArgumentConstraintTestBase
+    public class ArgumentConstraintTestBase
     {
-        internal IArgumentConstraint ConstraintField { get; set; }
-
-        protected abstract string ExpectedDescription { get; }
-
-        private IArgumentConstraint Constraint => this.ConstraintField;
+        public static Type[][] TestContextTypes =
+        {
+            new[] { typeof(CollectionContainsTests)},
+            new[] { typeof(CollectionIsEmptyTests)},
+        };

```

Comment:
If we have reflection, we could instead find subclasses of `ArgumentConstraintTestBase<IEnumerable<object>>` or whatever base test context we want to provide.